Src/VimCore/CoreInterfaces.fs

```diff
@@ -4532,8 +4534,19 @@ and SwitchModeEventArgs
     /// has no previous one
     member x.PreviousMode = _previousMode
 
+and MarkChangedEventArgs (_mark: Mark, _textBuffer: ITextBuffer) = 
+    inherit System.EventArgs()
+
+    member x.Mark = _mark
+    member x.TextBuffer = _textBuffer
+
+    override x.ToString() = _mark.ToString()
+
 and IMarkMap =
 
+    /// Raise the mark set event
+    abstract RaiseMarkSet: mark: Mark -> textBuffer: ITextBuffer -> unit

```

Comment:
This method feels like a bit of a leaky abstraction on where we chose to store the various mark values. I'm wondering if we need to centralize mark storage into local marks (per buffer) and global. Then have the mark changed events on those locations. That would in theory remove the need for this function and let the owner of the data raise the event when it changes.

Reply:
My precedent for this was `VimBuffer.RaiseErrorMessage` which is used only by `StatusUtil` in a "friendly" fashion.

The only two external callers of the `RaiseMarkSet` function are the vim text buffer and the jump list and we would limit the callers who could call it I knew how!

It's true that the vim text buffer could raise its own separate event, which is arguably better anyway because the consumer is already tied to a vim buffer (one mark glyph tagger per buffer).

Then the jump list could delegate the storage of the "last jump location" to the vim text buffer.

Reply:
It turns out that the jump list is scoped to a window so we have three different kind of marks: global, local and window.